.github/actions/build-sign-publish-chainlink/action.yml

```diff
@@ -151,29 +147,46 @@ runs:
       cosign-release: 'v1.4.0'
 
   - if: inputs.sign-images == 'true'
-    name: Write signing key to disk (only needed for `cosign sign --key`)
+    name: Sign the published root Docker image
     shell: sh
-    run: echo "${{ inputs.cosign-private-key }}" > cosign.key
+    env:
+      COSIGN_EXPERIMENTAL: 1
+    run: |
+      IMAGES_NAME_RAW=${{ fromJSON(steps.buildpush-root.outputs.metadata)['image.name'] }}
+      IMAGE_NAME=$(echo "$IMAGES_NAME_RAW" | cut -d"," -f1)
```

Comment:
Can we set these in `GITHUB_ENV` in a previous step so we're not repeating it 3x times?